src/main/java/com/github/msemys/esjc/EventStoreTcp.java

```diff
@@ -291,6 +290,22 @@ public Transaction continueTransaction(long transactionId, UserCredentials userC
         return result;
     }
 
+    @Override
+    public CompletableFuture<AllEventsFilteredSlice> readAllEventsForwardFiltered(
+        Position position,
+        int maxCount,
+        int maxSearchWindow,
+        boolean resolveLinkTos,
+        Iterable<String> allowedEventTypes,
+        UserCredentials userCredentials) {
+
+        checkArgument(BATCH_SIZE_RANGE.contains(maxCount), "maxCount is out of range. Allowed range: %s.", BATCH_SIZE_RANGE.toString());
```

Comment:
Shouldn't there be a check for maxSearchWindow as well?
maxSearchWindow >= maxCount

Reply:
I don't think this check is necessary. The server does not enforce this contraint. If you want 50 events and set a window of 5 you will get  5 events or less. Not very sensible, but also not illegal.